packages/generator/src/edmx-to-vdm/common/action-function-return-types.ts

```diff
@@ -150,15 +150,27 @@ function getEdmReturnType(
 
 function getEntityReturnType(
   isCollection: boolean,
-  entity: VdmEntity
+  entities: VdmEntity[]
 ): VdmFunctionImportReturnType {
-  return {
-    returnTypeCategory: VdmReturnTypeCategory.ENTITY,
-    returnType: entity.className,
-    builderFunction: entity.className,
-    isMulti: isCollection,
-    isCollection
-  };
+  if (!entities.length) {
+    throw Error('Cannot get entity return type from an empty collection.');
+  }
+
+  return entities.length === 1
+    ? {
+        returnTypeCategory: VdmReturnTypeCategory.ENTITY,
+        returnType: first(entities)!.className,
+        builderFunction: first(entities)!.className,
+        isMulti: isCollection,
+        isCollection
+      }
+    : {
+        returnTypeCategory: VdmReturnTypeCategory.ENTITY_NOT_DESERIALIZABLE,
+        returnType: 'any',
```

Comment:
Maybe `void` is the better return type here, given that nothing will ever be returned ;)